### PR TITLE
(#3488) - get rid of manualDescEnd

### PR DESCRIPTION
**Pull request by nolanlawson**
Seems to be passing in Firefox and Chrome. Better
wait to make sure IE works.

### CODE REVIEW

File: lib/adapters/idb/idb.js [hunk 1]
```diff
@@ -367,54 +367,56 @@ function init(api, opts, callback) {
     };
   };
 
-  function allDocsQuery(totalRows, opts, callback) {
```
Comment (nolanlawson) [inline lib/adapters/idb/idb.js]: The diff is a little misleading. I moved a lot of logic into its own function, to avoid try/catch deoptimizations.

**Comment by nolanlawson** [comment]
To be safe, I'd also like to manually run the map/reduce tests on this, so please don't merge until someone has done that.

**Comment by daleharvey** [comment]
Cool, all passing and looks good to me, :+1: for when you are ready

**Comment by nolanlawson** [comment]
wait for this to be green: https://github.com/pouchdb/mapreduce/pull/259

**Comment by daleharvey** [comment]
Gonna just merge this after the release, its a cleanup that shouldnt change behaviour, but just in case it does :)

**Comment by nolanlawson** [comment]
Yup, better safe than sorry.

**Comment by daleharvey** [comment]
ok release done, safe now - https://github.com/pouchdb/pouchdb/commit/904c812b225ba8250cfd433a5ce74c20354ae030

**Pull request closed without merging**